src/database/volatile/mod.rs

```diff
@@ -55,3 +67,68 @@ impl Tabular for Database<'_> {
         todo!()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    /// A maximally simple Record implementation, consisting of only a BitVec value
+    /// Created for test purposes
+    pub struct Rec {
+        value: BitVec<u8, Msb0>,
+    }
+
+    impl Rec {
+        pub fn initialize(val: BitVec<u8, Msb0>) -> Self {
+            Self { value: val.clone() }
+        }
+    }
+
+    impl Record for Rec {
+        fn raw(&self) -> &BitSlice<u8, Msb0> {
+            return &self.value[..];
+        }
+    }
+
+    /// This test
+    /// - creates an example state test_state and Record test_rec
+    /// - checks that that test_state is initially not mapped in the database
+    /// - puts test_rec in the database, with test_state as its key
+    /// - checks that test_state now maps to test_rec.
+    #[test]
+    fn put_data_and_get_it() {
+        let mut db: Database = Database::initialize();
+        let test_state: State = 7;
+        assert!(db.get(test_state).is_none());
+        let test_rec: Rec = Rec::initialize(BitVec::<u8, Msb0>::new());
+        db.put(test_state, &test_rec);
+        if let Some(result_rec) = db.get(test_state) {
+            assert!(result_rec == test_rec.raw());
+        } else {
+            assert!(1 == 0);
```

Comment:
use `assert_eq!` when asserting equality comparisons. 

this prints a more helpful message (i.e., "assertion failed, LHS: X, RHS: Y" vs "assertion failed")